### PR TITLE
cutadapt: import from brewsci/science

### PR DESCRIPTION
**Pull request by suujia**

### CODE REVIEW

**Comment by suujia** [comment]
```
/usr/include/python2.7 -c cutadapt/_align.c -o build/temp.linux-x86_64-2.7/cutadapt/_align.o
    cutadapt/_align.c:12:20: fatal error: Python.h: No such file or directory
     #include "Python.h"
                        ^
    compilation terminated.
    gcc-4.8: internal compiler error: Segmentation fault (program as)
Failed to execute: python

```

**Comment by tseemann** [comment]
It says 2.7 or >3.3  
https://cutadapt.readthedocs.io/en/stable/installation.html#dependencies

**Comment by sjackman** [comment]
CircleCI on Linux
```
==> python3 -s /tmp/cutadapt--homebrew-virtualenv-20180808-7333-105ihyh/target/bin/virtualenv -p python3 /home/linuxbrew/.linuxbrew/Cellar/cutadapt/1.16/libexec
.
Error: Will not overwrite /home/linuxbrew/.linuxbrew/Cellar/cutadapt/1.16/libexec/lib/python3.7/site-packages/homebrew_deps.pth
==> FAILED
```
https://circleci.com/gh/brewsci/homebrew-bio/3212

CircleCI on macOS
```
    ERROR: Cython is not installed. Install at least Cython version 0.24 to continue.
```
https://circleci.com/gh/brewsci/homebrew-bio/3211

**Comment by suujia** [comment]
I can add cython dependency for macOS. 

Locally when I change it to python@2 it produces: 
```
python@2
-c
import setuptools, tokenize
__file__ = 'setup.py'
exec(compile(getattr(tokenize, 'open', open)(__file__).read()
  .replace('\r\n', '\n'), __file__, 'exec'))
--no-user-cfg
install
--prefix=/tmp/cutadapt--homebrew-virtualenv-20180809-14673-19d19vr/target
--single-version-externally-managed
--record=installed.txt
```
and doesn't seem to actually install

**Comment by sjackman** [comment]
`python@2` is the name of the package, but the executable has a different name. The name of the package and executable is…

Python 2: package is `python@2` and executable is `python2`
Python 3: package is `python` and executable is `python3.

**Comment by suujia** [comment]
```
Will not overwrite /home/linuxbrew/.linuxbrew/Cellar/cutadapt/1.16/libexec/lib/python3.7/site-packages/homebrew_deps.pth
```

File: Formula/cutadapt.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 "b64575241f64f6ec005a4d4137339fb0ba5e156e826db2fdb5f458060d9979e0"
+  end
+
+  resource "xopen" do
+    url "https://files.pythonhosted.org/packages/53/1f/01d32269f01fccf1a6d8d5d795f939cb56274c82f7520cc25152ee8de486/xopen-0.2.1.tar.gz"
+    sha256 "9b054f8c1c906ca416412e8b7430bac4e683a2c5ce1a59e7e62d667418165dfe"
+  end
+
+  def install
+    virtualenv_create(libexec, "python3")
```
Comment (sjackman) [inline Formula/cutadapt.rb]: Try removing this line. I believe it's redundant with the following line.
```ruby
virtualenv_create(libexec, "python3")
```

**Comment by suujia** [comment]
CI 
```
/usr/local/Cellar/cutadapt/1.16/bin/cutadapt -h 2>&1
<0> expected but was <1>.
```
Locally: 
```
/home/linuxbrew/.linuxbrew/Cellar/cutadapt/1.14/libexec/bin/python3.6: error while loading shared libraries: libpython3.6m.so.1.0: cannot open shared object file: No such file or directory
```

Maybe I should add (?): 
```
    xy = Language::Python.major_minor_version "python3"
    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
```

**Comment by sjackman** [comment]
These lines shouldn't be necessary
```ruby
    xy = Language::Python.major_minor_version "python3"
    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
```
because setting `PYTHONPATH` is handled by `virtualenv_install_with_resources`

**Comment by sjackman** [comment]
You can troubleshoot the `brew test cutadapt` failure interactively in `linuxbrew/linuxbrew`. I see
```
cutadapt -h
…
pkg_resources.DistributionNotFound: The 'xopen>=0.3.2' distribution was not found and is required by cutadapt
```

**Comment by sjackman** [comment]
This formula is installing `xopen` 0.2.1. `cutadapt` requires 0.3.2 or newer. (See above)

**Comment by suujia** [comment]
@sjackman thanks for the help! :)

**Comment by sjackman** [comment]
Merged. Thanks again, Susie! 🏆

**Pull request closed without merging**